hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/functional/TestHoodieSparkMergeOnReadTableClustering.java

```diff
@@ -60,20 +61,30 @@ class TestHoodieSparkMergeOnReadTableClustering extends SparkClientFunctionalTes
 
   private static Stream<Arguments> testClustering() {
     return Stream.of(
-        Arguments.of(true, true, true),
-        Arguments.of(true, true, false),
-        Arguments.of(true, false, true),
-        Arguments.of(true, false, false),
-        Arguments.of(false, true, true),
-        Arguments.of(false, true, false),
-        Arguments.of(false, false, true),
-        Arguments.of(false, false, false)
-    );
+        Arrays.asList(true, true, true),
+        Arrays.asList(true, true, false),
+        Arrays.asList(true, false, true),
+        Arrays.asList(true, false, false),
+        Arrays.asList(false, true, true),
+        Arrays.asList(false, true, false),
+        Arrays.asList(false, false, true),
+        Arrays.asList(false, false, false))
+        .flatMap(arguments -> {
+          ArrayList<Boolean> enableRowClusteringArgs = new ArrayList<>();
+          enableRowClusteringArgs.add(true);
+          enableRowClusteringArgs.addAll(arguments);
+          ArrayList<Boolean> disableRowClusteringArgs = new ArrayList<>();
+          disableRowClusteringArgs.add(false);
```

Comment:
Appreciate your intent, but if we'd be chaining every parameter like that this could would become unreadable. Let's just add one more column for parameter (and add a comment at the top enumerating all the params in this matrix)